### DISCUSSION TITLE
Prepare dependencies of current package before running a script?

### DISCUSSION THREAD

**Discussion by elado** [nrwl/nx#9668] [General]
What's the nx way to "prepare dependencies" of a package?

I want to keep my existing yarn scripts, e.g. `"start": "next dev"` but want to ensure that all the dependencies of the current package are built. So something like `"start": "nx prepare @scope/app:start && next dev"`. Technically I expect it to satisfy the environment according to the defined `targetDependencies` of the current script, without running the script on the current package.

This would mostly fit uncachable scripts like starting a server or running ad-hoc lint/test with specific files, or test watchers.

```json
  "targetDependencies": {
    "build": [
      {
        "target": "build",
        "projects": "dependencies"
      }
    ],
    "start": [
      {
        "target": "build",
        "projects": "dependencies"
      }
    ]
  },
```

**Comment by fregante** [comment]
`build-deps` seems to be the right command, if your apps has it. This is not very well documented: https://nx.dev/technologies/typescript/recipes/switch-to-workspaces-project-references#register-nx-typescript-plugin
